lib/rust/frp/src/network.rs

```diff
@@ -99,25 +103,41 @@ impl Network {
     /// network, like animation instances.
     pub fn store<T: 'static + CloneRef>(&self, item: &T) {
         let item = item.clone_ref();
-        self.data.storage.borrow_mut().push(Box::new(item));
+        self.store_boxed(Box::new(item));
+    }
+
+    /// Store an arbitrary boxed item.
+    ///
+    /// Force the compiler to never inline this function, so the generated code (including two panic
+    /// handlers) can be shared across all registered items independent of their type.
+    #[inline(never)]
+    fn store_boxed(&self, item: Box<dyn Any>) {
+        self.data.storage.borrow_mut().push(item);
     }
 
     /// Register the node and return it's weak reference.
     pub fn register_raw<T: HasOutputStatic>(&self, node: stream::Node<T>) -> stream::WeakNode<T> {
         let weak = node.downgrade();
-        let node = Box::new(node);
-        self.data.nodes.borrow_mut().push(node);
+        self.register_boxed(Box::new(node));
         weak
     }
 
     /// Register the node and return a new `Stream` reference.
     pub fn register<Def: HasOutputStatic>(&self, node: stream::Node<Def>) -> Stream<Output<Def>> {
         let stream = node.clone_ref().into();
-        let node = Box::new(node);
-        self.data.nodes.borrow_mut().push(node);
+        self.register_boxed(Box::new(node));
         stream
     }
 
+    /// Register a boxed node. Non-generic part of registration.
+    ///
+    /// Force the compiler to never inline this function, so the generated code (including two panic
+    /// handlers) can be shared across all registered nodes independent of their type.
+    #[inline(never)]
+    fn register_boxed(&self, node: Box<dyn Item>) {
+        self.data.nodes.borrow_mut().push(node);
+    }
```

Comment:
How does it affect the compile time? Does it really make a difference?

Reply:
This is the same story as `store_boxed`, as both were introduced togerher. The registration is the more important of them though, as it is called per each created node, thus the previous borrow/push needed to be compiled separately for each individual node type in the codebase.

Reply:
This is not so important to me as the previous ne as this is only when creating nodes, not firing them, so I'm totallyy OK with leaving it this way :)